vignettes/build-a-basic-package.Rmd

```diff
@@ -0,0 +1,936 @@
+---
+title: "Build a Basic Package"
+author: "Jennifer Bryan"
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    mathjax: null
+    fig_caption: false
+    toc: true
+    toc_float:
+      collapsed: false
+      smooth_scroll: false
+    toc_depth: 3
+vignette: >
+  %\VignetteIndexEntry{Build a Basic Package}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+params:
+   debug: FALSE
+   github: FALSE
+---
+
+```{r setup, include = FALSE, cache = FALSE}
+knitr::opts_chunk$set(error = TRUE, collapse = TRUE, comment = "#>")
+```
+
+## Load devtools
+
+Load the devtools package.
+
+```{r}
+library(devtools)
+```
+
+Since devtools is a workflow package, you may consider loading it automatically in all of your interactive R sessions. In general, it's not a great idea to load packages this way, as it invites you to create R scripts that don't reflect all of their dependencies via explicit calls to `library(foo)`. But devtools is meant to smooth the process of package development and is, therefore, unlikely to get baked into any analysis scripts. The pros may outweigh the cons in this case.
+
+You can do this by adding these lines to your `.Rprofile`:
+
+```{r eval = FALSE}
+if (interactive()) {
+  suppressMessages(require(devtools))
+}
+```
+
+## Toy package: foofactors
+
+We use multiple functions from devtools to build a small toy package from scratch, with features commonly seen in released packages:
+
+  * Developed under version control (Git, in this case) and in the open (GitHub, in this case). *optional, but shown*
+  * Documentation for individual functions, using convenience functions in devtools that call [roxygen2](https://CRAN.R-project.org/package=roxygen2).
+  * Unit testing, using convenience functions in devtools to exploit [testthat](https://CRAN.R-project.org/package=testthat).
+  * Documentation for the package as a whole, via a package-level help file, a vignette, and an executable `README.Rmd`.
+
+We call the package **foofactors** and it will consist of a couple functions for handling factors. Please note that these functions are super simple and definitely not the point! For a proper package for factor handling, please see [forcats](https://cran.r-project.org/package=forcats).
```

Comment:
I think it's fine to stick with this example.  Any sufficiently motivating example here will inevitably end up in a package.